### PR TITLE
Researchable Smartmines + Exploding Rubber Duck traitor item

### PR DESCRIPTION
**Pull request by WhyIsCaeciliusTaken**
## About The Pull Request

This PR significantly changes how landmines work in game, and can be summarised in three parts:

**1: Landmines can now be planted, moved and disarmed.**
Adds disarmed versions of most mines, which can be safely picked up and carried around, and walking over one does nothing. Using a disarmed mine on a valid (ie not space, a wall or lava/chasms)  turf will plant and arm the mine after a short windup (3 seconds for most mines), after which it functions normally. Mines can be disarmed with a multitool, which takes 20 seconds by default, and drops the appropriate disarmed mine (or the base dummy mine if it's a mine type you shouldn't have like a bwoink mine). Clown mines take 6 times as long to disarm to make them more annoying,  and are now louder when set off. Stun mines now do stamina damage instead of hardstunning.

**2: New mines and mine tech.**
Adds three new tech nodes, three new mines for security, and allows you to build both the new mines as well as stun and clown mines. Nonlethal Landmine Technology allows sec to build stun mines, and requires basic security equipment as a prerequisite, as well as costing 1000 points.

Once you have researched industrial engineering, weapons development and nonlethal mines, you can research Smart Landmine Technology for 2500 points, which allows you to build smart stun mines. These mines are more expensive but are harder to disarm (25s) and do not trigger if the person walking over them is mindshielded, opening up interesting tactical possibilities for security.

Once you have smartmines you can research Advanced Landmine Technology for 2500, which gives you two more advanced smartmines, the rapid deployment and the  Sledgehammer. The RD is easier to disarm (12s) but has a very short arm time (1s), allowing for mid combat placement if you play it right, allowing sec to, say, enter through a door and deploy one behind them to block escape via that route. It also fits in pockets and boxes, unlike the other mines.

The sledgehammer is slower to arm (5s) but is very hard to disarm (35s), does extra stamina damage as well as 40 brute, making it more suited to defensive use such as blocking off the armoury.

The lower tier mines can be ordered from cargo at a reasonably high cost.

**3: Exploding Rubber Duck:**
Adds two further new mine types, the regular and high yield exploding rubber duck. Both are explosive, take a very long time to disarm (40s) and look like a rubber duck when armed. The high yield version is far more powerful but costs more (10TC vs 4TC). Both make the same sound effect as a rubber duck before detonation, d'aww!

## Why It's Good For The Game
Mines are basically nonexistant currently, and their new abilities to be moved, disarmed and discriminate between friend and foe open up interesting oppertunities for area denial or other tactics  (such as blocking retreat as mentioned above).

Printable clown mines give clowns a clown tech item to play with that does not require 8 million bananium.

The exploding rubber ducks are just plain funny and fun to use.
## Changelog
:cl:
add: Added ability to move and plant mines, as well as disarm them with a multitool
add: Added ability for mines to be "smart" and detect mindshield implants if a particular mine type
add: Clown and stun mines can now be printed in the service and security protolathes respectively
add: three new mine related tech nodes, and three new printable smart mine types for security
add: Added the exploding rubber duck as a 4TC traitor and clownops item, with a more powerful version available for 10TC.
add: Some mines can be ordered from cargo
tweak: stun mines now do stamina damage instead of a hardstun
/:cl:

### CODE REVIEW

File: code/game/objects/effects/mines.dm [hunk 1]
```diff
@@ -63,32 +212,38 @@
...
+	disarm_product = /obj/item/deployablemine/sleepy
 
 /obj/effect/mine/sound
 	name = "honkblaster 1000"
 	var/sound = 'sound/items/bikehorn.ogg'
+	var/volume = 150
+	disarm_time = 1200 // very long disarm time to expand the annoying factor
+	disarm_time = /obj/item/deployablemine/honk
 
 /obj/effect/mine/sound/mineEffect(mob/victim)
```
Comment (zeskorion) [inline code/game/objects/effects/mines.dm:241]: Perhaps this should call a Soundbang?

File: code/game/objects/effects/mines.dm [hunk 2]
```diff
@@ -63,32 +212,38 @@
...
-	playsound(loc, sound, 100, 1)
+	playsound(loc, sound, volume, 1)
 
 
 /obj/effect/mine/sound/bwoink
```
Comment (zeskorion) [inline code/game/objects/effects/mines.dm:250]: I think these should totally be disarmable and obtainable. It'd be great if you could put an attackby in the sound mines to check if hit by a sound synth, and then transfer the sound the synth was set to to the mine? That part's just a suggestion though.

**Comment by bloons3** [comment]
Any plan for a syndicate version that checks faction?
Ops don't have any sort of effective area denial weapon.

**Comment by WhyIsCaeciliusTaken** [comment]
> Any plan for a syndicate version that checks faction?
> Ops don't have any sort of effective area denial weapon.

Currently no, but with a bit of modification to the smartmine proc to check for implants instead of TRAIT_MINDSHIELD it certainly could be done.

**Comment by Isy232** [comment]
Perhaps put it behind basic weapons tech instead of advanced, make a crate so they are order-able and add a few of these to the armory.

Otherwise I doubt we will see them at all.

File: code/game/objects/effects/mines.dm (hunk 1 above)
Comment (WhyIsCaeciliusTaken) [reply code/game/objects/effects/mines.dm:241]: The idea is that they're completely harmless and just minorly annoying, but I can make them actually disruptive (and therefore more expensive) if you'd like.

File: code/game/objects/effects/mines.dm (hunk 2 above)
Comment (WhyIsCaeciliusTaken) [reply code/game/objects/effects/mines.dm:250]: That sounds hilarious. I'll give it a try.

**Review by zeskorion — APPROVED** [review]

**Pull request merged**